snapcraft/plugins/_python/errors.py

```diff
@@ -47,3 +47,19 @@ class MissingSitePyError(PythonPluginError):
 
     def __init__(self, site_py_glob):
         super().__init__(site_py_glob=site_py_glob)
+
+
+class PipListInvalidJsonError(PythonPluginError):
+
+    fmt = "Pip packages output isn't valid json: {json!r}"
```

Comment:
Here again, it feels like you are being extremely cautious. If pip list --json doesn't return a json, I think it's time to panic because everything is likely to be broken. In many places, we are not wrapping this unexpected errors, and I'm not sure if we should. But that's a good discussion to have.